File: scalechain-proto-codec/src/main/scala/io/scalechain/blockchain/proto/codec/privateparts_codecs.scala
package io.scalechain.blockchain.proto.codec

import io.scalechain.blockchain.proto._
import io.scalechain.blockchain.proto.codec.primitive.{FixedByteArray, Bool, VarStr, BigIntForLongCodec}
import scodec.Codec
import scodec.codecs._

object BlockConsensusCodec extends MessagePartCodec[BlockConsensus] {
  val codec : Codec[BlockConsensus] = {
    ("header" | BlockHeaderCodec.codec) ::
    ("height" | int64L )
  }.as[BlockConsensus]
}

File: scalechain-proto-codec/src/main/kotlin/io/scalechain/blockchain/proto/codec/privateparts_codecs.kt
package io.scalechain.blockchain.proto.codec

import io.scalechain.blockchain.proto.*
import io.scalechain.blockchain.proto.codec.primitive.*

object BlockConsensusCodec : Codec<BlockConsensus> {
  override fun transcode(io : CodecInputOutputStream, obj : BlockConsensus? ) : BlockConsensus? {
    val header = BlockHeaderCodec.transcode(io, obj?.header)
    val height = Codecs.Int64L.transcode(io, obj?.height)

    if (io.isInput) {
      return BlockConsensus(
        header!!,
        height!!
      )
    }
    return null
  }
}
